packages/create-redwood-app/template/.vscode/launch.json

```diff
@@ -21,6 +21,32 @@
       "remoteRoot": "${workspaceFolder}/node_modules/@redwoodjs/api-server/dist",
       "sourceMaps": true,
       "restart": true
-    }
+    },
+    {
+      "command": "yarn redwood test api",
```

Comment:
Is there a specific reason to separate api and web?

I'm wondering what the best solution for debugging tests is really with these extensions. Maybe the "debug" button on top of the tests that we get with the extension? 

Expanded a bit here: https://s.tape.sh/cPLS9LV6

Reply:
> Is there a specific reason to separate api and web?

The assumption is when debugging you are typically in one context or the other but I suppose in watch you could filter from there.

Reply:
> I'm wondering what the best solution for debugging tests is really with these extensions. Maybe the "debug" button on top of the tests that we get with the extension?

I think this is the typical flow -- the debugging attachment is nice if you were running a bunch of tests at once and say sticking a breakpoint in your application code.

Reply:
To crystallize this, there are a few benefits to using the launch config:
1. you can make use of watch mode with the debugger which is nice, you defacto always have the debugger enabled and can auto rerun failing tests trivially instead of "focus" mode on a single test -- this let's you jump around and go through several tests a seamlessly vs. click debug and work on one test, then do the same for another test -- whether both are necessary depends on assumptions on workflow
2. given the above you typically are working on one "side" or another -- so assuming the workflow of (1) has value then it makes sense to me to split the tests -- though you can always filter down further in the CLI so this is a weak conviction item

Reply:
> If there was a nice RW command for running in debug mode, I'd say let's just delete the launch config but I don't see an easy way to do it since you need to pass some args to node.

We've got `yarn rw dev` already setup that way, and setting up Jest looks much the same: https://jestjs.io/docs/troubleshooting#debugging-in-vs-code.

I kinda feel like we should prefer `attach` to `launch`, but that may just be a personal preference. But it does sound like it'd enable the same workflow you described here: 

> this let's you jump around and go through several tests a seamlessly vs. click debug and work on one test

If we configure for `attach`, you can run tests in watch mode in another terminal even, and just attach to debug when you need to.